src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,192 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int CAPACITY = 16;
+    private int size = 0;
+    private Node<T> header;
+    private Node<T> currentNode;
+    private Node<T> first;
```

Comment:
@gkhrshch what is the conceptual difference between first and header? These variable names could be misunderstanding. Let's call them header and tail